sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/RowUtils.scala

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.execution.python
+
+import java.io._
+
+import com.google.common.io.Closeables
+
+import org.apache.spark.io.NioBufferedFileInputStream
+import org.apache.spark.memory.{MemoryConsumer, SparkOutOfMemoryError, TaskMemoryManager}
+import org.apache.spark.{SparkEnv, SparkException}
+import org.apache.spark.serializer.SerializerManager
+import org.apache.spark.sql.catalyst.expressions.UnsafeRow
+import org.apache.spark.unsafe.Platform
+import org.apache.spark.unsafe.memory.MemoryBlock
+
+// The whole file is copied from Spark `RowQueue` to expose row queue utils to rapids execs
+
+/**
+ * A RowQueue is an FIFO queue for UnsafeRow.
+ *
+ * This RowQueue is ONLY designed and used for Python UDF, which has only one writer and only one
+ * reader, the reader ALWAYS ran behind the writer. See the doc of class BatchEvalPythonExec
+ * on how it works.
+ */
+private[sql] trait RowQueue {
```

Comment:
I personally would rather not copy this code over into our package, but use TrampoineUtil if necessary to get access to it.

Reply:
Googd to know. Will learn the `TrampolineUtil ` and added it as a TODO.
I could imagine this code will be no longer needed if we supported columnar way in the future.